### PR TITLE
Feat: migrate to Github Actions

### PR DESCRIPTION
**Pull request by Fabious**
Related: #31

### CODE REVIEW

**Comment by Fabious** [comment]
Actions preview:

https://github.com/Fabious/react-esi/actions

**Comment by dunglas** [comment]
We should be able to use the coveralls GitHub Action to upload the coverage.

**Comment by Fabious** [comment]
Ho ok, got an error with coveralls as seen here : https://github.com/Fabious/react-esi/runs/4162385127?check_suite_focus=true

Same error on local (main branch), will try to find a solution

**Comment by Fabious** [comment]
ok fixed coveralls: https://github.com/Fabious/react-esi/runs/4163101844?check_suite_focus=true

Can you check if it's all good? 
After that the last thing is to automate publishing to npm?

**Comment by Fabious** [comment]
I added a release action, for this one i'm not sure how to test it and how we should implement the said release
If you have the time to look =]

**Comment by Fabious** [comment]
As this PR has gone stale (2021 😵‍), i leave this here as a CLI helper exists and i think it will help

https://github.com/semantic-release/semantic-release/blob/master/docs/usage/getting-started.md#getting-started

Maybe I'll look into it and this PR when i have a bit of time!

**Comment by Fabious** [comment]
I have simplified this PR : I initiated only one but important Action for testing the app on a PR

Will make an other PR for the release phase (aka publishing on npm)

File: .github/test.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Tests / Lint
+
+on: [pull_request]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16]
```
Comment (dunglas) [inline .github/test.yml:10]:
```suggestion
        node-version: [16, 18, 20]
```

?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,5 @@
 # React ESI: Blazing-fast Server-Side Rendering for React and Next.js
 
-[![Build Status](https://travis-ci.org/dunglas/react-esi.svg?branch=master)](https://travis-ci.org/dunglas/react-esi)
```
Comment (dunglas) [inline README.md:3]: We could replace it by the GHA badge!

**Review by dunglas — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Fabious) [reply README.md:3]: good idea =]

**Comment by dunglas** [comment]
Thank you Fabien!

**Pull request merged**